onnxruntime/core/graph/graph.cc

```diff
@@ -1257,13 +1257,22 @@ void Graph::ReverseDFSFrom(const std::vector<NodeIndex>& from,
     node_vec.push_back(GetNode(i));
   }
 
-  ReverseDFSFrom(node_vec, enter, leave, comp);
+  ReverseDFSFrom(node_vec, enter, leave, comp, {});
 }
 
 void Graph::ReverseDFSFrom(const std::vector<const Node*>& from,
                            const std::function<void(const Node*)>& enter,
                            const std::function<void(const Node*)>& leave,
                            const std::function<bool(const Node*, const Node*)>& comp) const {
+
+  ReverseDFSFrom(from, enter, leave, comp, {});
+}
+
+void Graph::ReverseDFSFrom(const std::vector<const Node*>& from,
```

Comment:
I saw a UT under ir/graph_test for ReverseDFS. Is it possible to add similar UT for the new method under orttraining?

Reply:
Yes, will add similar UT.

Reply:
I saw another place might be better for the UT - GradientGraphBuilderTest->TrainingSession_BertToy, perhaps duplicate that one and make your own if you like. 

---
In reply to: [434769892](https://github.com/microsoft/onnxruntime/pull/4116#discussion_r434769892) [](ancestors = 434769892)

Reply:
So I ended up modifying GraphTest.ReverseDFS. The _stop_ method modifies its traversal behavior.